tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/Package/ConfigBasedTool.cs

```diff
@@ -0,0 +1,354 @@
+using Azure.Sdk.Tools.Cli.Commands;
+using Azure.Sdk.Tools.Cli.Helpers;
+using Azure.Sdk.Tools.Cli.Models;
+using Microsoft.Extensions.Logging;
+
+namespace Azure.Sdk.Tools.Cli.Tools.Package;
+
+/// <summary>
+/// Base class for tools that need configuration-based operations, process execution, and common response patterns.
+/// Provides shared functionality for configuration management, validation, response creation, and process execution.
+/// </summary>
+public abstract class ConfigBasedTool : MCPTool
```

Comment:
Seems like this class consists mainly of helper functions. Maybe it would be beneficial to convert the helpers into a helper class so that any helpers can be used without needing to inherit from `ConfigBasedTool`?

Reply:
We do have `specGenSdkConfigHelper` for config parsing/substitution, and the choice to add `ConfigBasedTool` was intentional to remove a lot of repeated boilerplate across five tools (`update-changelog, update-metadata, update-ci, update-version, build`). The base class centralizes the Process orchestration, shared validation, standardized responses.